discovery_engine_core/core/src/engine.rs

```diff
@@ -1462,6 +1472,8 @@ pub(crate) mod tests {
                     .unwrap_or(u32::MAX),
                 de_config: None,
                 log_file: None,
+                data_dir: "tmp_test_data_dir".into(),
+                use_in_memory_db: true,
```

Comment:
At some point we probably need to setup some testing with a `TempDir` `data_dir` and proper teardown/setup but that's outside of the scope of this PR but I might need to do it for TY-3061.